source/fasta/fasta.d

```diff
@@ -113,6 +113,12 @@ auto tokenParser(Chain, char header = '>', char fieldsep = '|')(Chain c) if (isI
             if(pos == chain.window.length)
                 // reaches the end of the stream
                 return FastaToken.init;
+            // deal with empty lines at the beginning
+            while (chain.window[0 .. $].strip.length == 0)
+              {
+                pos += chain.window.length;
+                chain.extend(0);
+              }
```

Comment:
Don't do this on every token, as it only needs to be done at the beginning. Maybe do it outside the Result struct, in the function before you return it (just after extending for one line).

Reply:
Good idea!

Reply:
Steve @schveiguy: Moving the lines out of the `nextToken()` does not work if I keep storage type of `Result` as `static`. I need to pass in `lines` and starting `pos` Which can’t be determined at compile time.

So two possible choices:
- Remove `static` Flag and initialize pos with the position where emtpy lines ends. This will reduce performance.
- Somehow store current reading position internally in `Chain`. This means update `iopipe`.  I don’t know what’s the consequence of this.

Reply:
Thanks Steve. I did not know `release` can be used this way.